src/main/scala/org/ergoplatform/ErgoScriptPredef.scala

```diff
@@ -37,7 +37,7 @@ object ErgoScriptPredef {
   def expectedMinerOutScriptBytesVal(delta: Int, minerPkBytesVal: Value[SByteArray]): Value[SByteArray] = {
     val genericPk = ProveDlog(CryptoConstants.dlogGroup.generator)
     val genericMinerProp = rewardOutputScript(delta, genericPk)
-    val genericMinerPropBytes = ErgoTreeSerializer.DefaultSerializer.serializeWithSegregation(genericMinerProp)
+    val genericMinerPropBytes = DefaultSerializer.serializeErgoTree(genericMinerProp.treeWithSegregation)
```

Comment:
@greenhat I suggest rewardOutputScript should return ErgoTree. So the decision of whether to segregate constants or not and to which extent is done inside the method.

Reply:
Done.